TabloidCLI/UserInterfaceManagers/TagManager.cs

```diff
@@ -46,22 +51,81 @@ public IUserInterfaceManager Execute()
 
         private void List()
         {
-            throw new NotImplementedException();
+            List<Tag> tags = _tagRepository.GetAll();
+            foreach (Tag tag in tags)
+            {
+                Console.WriteLine(tag);
+            }
         }
 
+        private Tag Choose(string prompt = null)
+        {
+            if (prompt == null)
+            {
+                prompt = "Please choose a Tag:";
+            }
+
+            Console.WriteLine(prompt);
+
+            List<Tag> tags = _tagRepository.GetAll();
+
+            for (int i = 0; i < tags.Count; i++)
+            {
+                Tag tag = tags[i];
+                Console.WriteLine($" {i + 1}) {tag.Name}");
+            }
+            Console.Write("> ");
+
+            string input = Console.ReadLine();
+            try
+            {
+                int choice = int.Parse(input);
+                return tags[choice - 1];
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Invalid Selection");
+                return null;
+            }
+        }
         private void Add()
         {
-            throw new NotImplementedException();
+
+            Console.WriteLine("New Tag");
+            Tag tag = new Tag();
+
+            Console.Write("New Tag: ");
+            tag.Name = Console.ReadLine();
+
+            _tagRepository.Insert(tag);
         }
 
         private void Edit()
         {
-            throw new NotImplementedException();
+            Tag tagToEdit = Choose("Which tag would you like to edit?");
+            if (tagToEdit == null)
+            {
+                return;
+            }
+
+            Console.WriteLine();
+            Console.Write("New tag (blank to leave unchanged: ");
+            string tagName = Console.ReadLine();
+            if (!string.IsNullOrWhiteSpace(tagName))
+            {
+                tagToEdit.Name = tagName;
```

Comment:
Instead of doing an if statement do a while loop to make sure the user inputs a non null value